index.js

```diff
@@ -19,40 +17,76 @@ const getSecurityVulnerabilitiesForPhpDependency = async (token, dependency) =>
  * Get the list of security vulnerabilities affecting a given GitHub organization's
  * repositories.
  *
- * @param token {string} A GitHub personal access token
- * @param org {string} A GitHub organization's name
- * @returns {Promise<Object.<string, array>>}
+ * @param {string} token - A GitHub personal access token
+ * @param {string} org - A GitHub organization's name
+ * @returns {Promise<{identifiers, inspectedOn, packageName, repo, severity, vulnerableVersionRange}[]>}
  */
-module.exports.getSecurityVulnerabilitiesForGitHubOrg = async (token, org) => {
+const getSecurityVulnerabilitiesForGitHubOrg = async (token, org) => {
   console.log(`Scanning for vulnerabilities in GitHub org. ${org}`)
-  const results = {}
-  
-  const repos = await github.listRepos(org, token)
+  const repos = await github.listRepos(token, org)
+  return getSecurityVulnerabilitiesForGitHubRepos(token, repos)
+}
+
+/**
+ * Get a list of security vulnerabilities affecting the specified GitHub repos.
+ *
+ * Only repos whose dependencies could be determined will be included.
+ *
+ * @param {string} token - A GitHub personal access token
+ * @param {string[]} repos - A list of repo names
+ * @returns {Promise<{identifiers, inspectedOn, packageName, repo, severity, vulnerableVersionRange}[]>}
+ */
+const getSecurityVulnerabilitiesForGitHubRepos = async (token, repos) => {
+  const results = []
   for (let i = 0; i < repos.length; i++) {
     const repo = repos[i]
     const vulnerabilities = await getSecurityVulnerabilitiesForGitHubRepo(token, repo)
-    if (vulnerabilities === undefined) {
-      results[repo] = null
-      continue
-    }
-    results[repo] = []
-    for (let j = 0; j < vulnerabilities.length; j++) {
-      results[repo].push(vulnerabilities[j])
+    if (Array.isArray(vulnerabilities)) {
+      results.push(...vulnerabilities)
     }
   }
-  
   return results
 }
 
+/**
+ * Get the security vulnerabilities for the specified GitHub repo. It will
+ * return an array of vulnerabilities, or `undefined` if the GitHub repo's
+ * dependencies could not be determined.
+ *
+ * NOTE: The data returned about each vulnerability will include a `repo` field
+ * with given repo name.
+ *
+ * @param {string} token - A GitHub personal access token
+ * @param {string} repo - A GitHub repo (formatted as "owner/library")
+ * @returns {Promise<undefined | {identifiers, inspectedOn, packageName, repo, severity, vulnerableVersionRange}[]>}
+ */
 const getSecurityVulnerabilitiesForGitHubRepo = async (token, repo) => {
   console.log(`Looking for vulnerabilities in GitHub ${repo}`)
-  const results = []
-  
   const dependencies = await github.getDependenciesOfRepo(token, repo)
   if (dependencies === undefined) {
     console.log('(unable to determine dependencies)')
     return undefined
   }
+  const vulnerabilities = await getSecurityVulnerabilitiesForPhpDependencies(
```

Comment:
For logical consistency, it seems `Php` should be removed from this function name or added to `getDependenciesOfRepo`.

Reply:
Thanks for pointing that out. My (half-developed) thought about that is that I want to return all security vulnerabilities we can find for the repo (regardless of language), but that at the moment we only know how to find PHP dependencies/vulnerabilities.

If/when I add checks for, say, JavaScript dependencies and vulnerabilities, I think I'd want to include them in the results of calls to `getSecurityVulnerabilitiesForGitHubRepo()` by having it internally also call a `getSecurityVulnerabilitiesForJavaScriptDependencies()` method.

Does that make sense, or does it still seem like I should rename one of these two methods to synchronize them?

(When that time comes, it looks like I'll need to separate out the JS and PHP dependencies in order to pass them to the appropriate `getSecurityVulnerabilitiesFor___Dependencies()` function.)

Reply:
I guess it depends on the long-term plan for `getDependenciesOfRepo`. If it will return all dependencies, somehow grouped by language, then I guess nothing needs to change. But I had guessed that you would have multiple `getXxxDependenciesOfRepo` functions that each return dependencies for a given language that would be passed to the appropriate `getSecurityVulnerabilitiesForXxxDependencies` function. (Or have both functions be language-agnostic.)

Reply:
On further thought, you make some very good points. I'll rename some things here that are PHP-specific.